ge/org.osate.ge.ba/src/org/osate/ge/ba/ui/properties/BehaviorTransitionPropertySection.java

```diff
@@ -159,25 +308,35 @@ public void widgetSelected(final SelectionEvent e) {
 				};
 
 				final EditEmbeddedTextDialog dlg = new EditEmbeddedTextDialog(Display.getCurrent().getActiveShell(),
-						"Edit Transition Condition", "Enter new dispatch condition.", xtextAdapter, behaviorTransition,
-						isValidModification);
+						"Edit Transition Condition", "Enter new dispatch condition.", xtextAdapter,
+						SWT.BORDER | SWT.SINGLE, GridDataFactory.fillDefaults().indent(10, 0).grab(true, true).create(),
+						behaviorTransition, isValidModification);
 				if (dlg.open() == Window.OK) {
 					// Edit condition
 					BehaviorAnnexSelectionUtil.getActiveEditor().ifPresent(editorPart -> {
 						final ActionService actionService = Adapters.adapt(editorPart, ActionService.class);
 						final ModelChangeNotifier modelChangeNotifier = Objects.requireNonNull(
 								editorPart.getAdapter(ModelChangeNotifier.class),
 								"Unable to get model change notifier");
-						actionService.execute("Modify Behavior Transition Condition", ExecutionMode.NORMAL,
-								new EmbeddedTextModificationAction(editingDomain, xtextDocument, xtextResource,
-										modelChangeNotifier, project, dlg.getText(), conditionTextValue));
+						if (xtextDocument != null) {
+							// Execute modification with xtext document
+							actionService.execute("Modifying Behavior Transition Condition", ExecutionMode.NORMAL,
+									new EmbeddedTextModificationAction(xtextDocument, modelChangeNotifier, project,
+											dlg.getText(true)));
+						} else {
+							// Execute modification with xtext resource
+							actionService.execute("Modifying Behavior Transition Condition", ExecutionMode.NORMAL,
+									new EmbeddedTextModificationAction(editingDomain, xtextResource,
+											modelChangeNotifier, project, dlg.getText(false),
```

Comment:
getText(boolean) is also called with false here for editing the XtextResource (editor closed).